generator/src/main/java/com/graphicsfuzz/generator/knownvaluegeneration/ExpressionGenerator.java

```diff
@@ -486,19 +488,55 @@ private Expr generateVariableFact(Value value,
                                     Stmt stmtToInsertBefore) {
     boolean atGlobalScope = factManager.globalScope() || generator.nextBoolean();
     final String varName = genVarName(value, atGlobalScope);
-
+    final Expr initializer = generateExpr(atGlobalScope ? globalFactManager : factManager,
+        currentFunction, stmtToInsertBefore, value);
+    // If generating global scope variables, initialisation would be performed later in a global
+    // initializer function.
     final VariableDeclInfo variableDeclInfo = new VariableDeclInfo(varName, null,
-        new Initializer(generateExpr(atGlobalScope ? globalFactManager : factManager,
-            currentFunction,
-            stmtToInsertBefore,
-            value
-        )));
-
+        atGlobalScope ? null : new Initializer(initializer));
     final VariablesDeclaration variablesDecl = new VariablesDeclaration(value.getType(),
         variableDeclInfo);
     final VariableDeclFact variableDeclFact = new VariableDeclFact(variablesDecl,
         variableDeclInfo, value);
+
     if (atGlobalScope) {
+      // Searching for the already declared globals initializer's function prototype, if not found,
+      // we have to generate a new prototype, function definition and a function call expression at
+      // the top level of main method to invoke the initializer function.
```

Comment:
How about you change:

FunctionDefinition initGlobalsFunction

to:

Optional<FunctionDefinition> maybeInitGlobalsFunction

and drop the "orElse(null)".

Then you can do:

FunctionDefinition initGlobalsFunction;
if (maybeInitGlobalsFunction.isPresent()) {
  initGlobalsFunction = maybeInitGlobalsFunction.get();
} else {
  // code to make the new function, assigning it to initGlobalsFunction.
}